pypeit/core/fitting.py

```diff
@@ -439,15 +442,16 @@ def robust_fit(xarray, yarray, order, x2=None, function='polynomial',
     iIter = 0
     qdone = False
     this_gpm = np.copy(in_gpm)
-    mskcnt = np.sum(this_gpm)
+    #mskcnt = np.sum(this_gpm)
     #pypeitFit = None
     while (not qdone) and (iIter < maxiter):
         if np.sum(this_gpm) <= np.sum(order) + 1:
             msgs.warn("More parameters than data points - fit might be undesirable")
         if not np.any(this_gpm):
             msgs.warn("All points were masked. Returning current fit and masking all points. Fit is likely undesirable")
 
-        pypeitFit = PypeItFit(xval=xarray, yval=yarray, func=function, order=order, x2=x2, weights=weights, gpm=this_gpm.astype(int),
+        pypeitFit = PypeItFit(xval=xarray, yval=yarray, func=function, order=np.atleast_1d(order), x2=x2, weights=weights,
```

Comment:
I am not liking this casting very much. Better to write code with flexible arguments in my opinion.